google/cloud/bigquery/table.py

```diff
@@ -2942,6 +2953,123 @@ def __repr__(self):
         return "TimePartitioning({})".format(",".join(key_vals))
 
 
+class PrimaryKey:
+    """Represents the primary key constraint on a table's columns.
+
+    Args:
+        columns: The columns that are composed of the primary key constraint.
+    """
+
+    def __init__(self, columns: List[str]):
+        self.columns = columns
+
+    def __eq__(self, other):
+        if not isinstance(other, PrimaryKey):
+            raise NotImplementedError
+        return self.columns == other.columns
+
+
+class ColumnReference:
+    """The pair of the foreign key column and primary key column.
+
+    Args:
+        referencing_column: The column that composes the foreign key.
+        referenced_column: The column in the primary key that are referenced by the referencingColumn.
+    """
+
+    def __init__(self, referencing_column: str, referenced_column: str):
+        self.referencing_column = referencing_column
+        self.referenced_column = referenced_column
+
+    def __eq__(self, other):
+        if not isinstance(other, ColumnReference):
+            raise NotImplementedError
+        return (
+            self.referenced_column == other.referencing_column
+            and self.referenced_column == other.referenced_column
+        )
```

Comment:
For this portion of the mod (2985-2987), coverage.py indicates that we are not exercising this code fully.

We do have tests that reference ColumnReferences, but not necessarily in a way that fully exercises each line of this code.

Reply:
Thanks. I've added `column_references` to the `__eq__` in commit https://github.com/googleapis/python-bigquery/pull/1755/commits/da663b2a61615bb43bea737088b5bc90efc13e1d